pkg/cmd/roachtest/test_runner.go

```diff
@@ -1733,56 +1738,94 @@ func (r *testRunner) teardownTest(
 	return nil
 }
 
-// inspectArtifacts inspects node logs and attempts to write helpful triage
-// information to the test log and testRunner.githubMessage
+// inspectArtifacts inspects logs and attempts to write helpful triage
+// information to the test log and testRunner to be used in github issues.
 // This method is best effort and should not fail a test.
 // This method writes to both testLogger which is expected to be test.log and
 // t.L() which is test-teardown.log since inspectArtifacts is called after
 // teardownTest
 func (r *testRunner) inspectArtifacts(
-	ctx context.Context, t *testImpl, testLogger *logger.Logger,
-) error {
+	ctx context.Context, t *testImpl, c *clusterImpl, testLogger *logger.Logger,
+) (inspectArtifactsErr error) {
 
 	if t.Failed() || roachtestflags.AlwaysCollectArtifacts {
 		t.L().Printf("Attempting to gather node fatal level logs for triage.")
-		out, err := gatherFatalNodeLogs(t, testLogger)
+		fatalOut, err := gatherFatalNodeLogs(t, testLogger)
 		if err != nil {
-			return err
-		}
-		if out == "" {
+			// even if we encounter an error continue on, this is best effort
+			if joinErr := errors.Join(inspectArtifactsErr, err); joinErr != nil {
+				return joinErr
+			}
+		} else if fatalOut == "" {
 			t.L().Printf("No fatal level logs found.")
-			return nil
 		} else {
 			testLogger.PrintfCtx(ctx, "CockroachDB contains Fatal level logs. Up to the first 10 "+
-				"will be shown here. See node logs in artifacts for more details.\n%s", out)
-			t.appendGithubMessage(out)
-			return nil
+				"will be shown here. See node logs in artifacts for more details.\n%s", fatalOut)
+			t.appendGithubFatalLogs(fatalOut)
+		}
+
+		t.L().Printf("Attempting to gather ip node mapping")
+		ipNodeMapOut, err := gatherNodeIpMapping(t, c)
+		if err != nil {
+			if joinErr := errors.Join(inspectArtifactsErr, err); joinErr != nil {
```

Comment:
`err != nil` implies that `errors.Join(inspectArtifactsErr, err) != nil`. The nested if is redundant.